burn-book/src/basic-workflow/model.md

```diff
@@ -87,6 +178,44 @@ impl ModelConfig {
 }
 ```
 
+<details>
+<summary><strong>🦀 References</strong></summary>
+
+In the previous example, the `init()` method signature uses `&` to indicate that the parameter types
+are references: `&self`, a reference to the current receiver (`ModelConfig`), and
+`device: &B::Device`, a reference to the backend device.
+
+```rust, ignore
+pub fn init<B: Backend>(&self, device: &B::Device) -> Model<B> {
+    Model {
+        // ...
+    }
+}
+```
+
+References in Rust allow us to point to a resource to access its data without owning it. The idea of
+ownership is quite core to Rust and is worth
+[reading up on](https://doc.rust-lang.org/book/ch04-00-understanding-ownership.html).
+
+In a language like C, memory management is explicit and up to the programmer, which means it is easy
+to make mistakes. In a language like Java or Python, memory management is automatic with the help of
+a garbage collector. This is very safe and straightforward, but also incurs a runtime cost.
+
+In Rust, memory management is rather unique. Aside from
+[primitive types](https://doc.rust-lang.org/rust-by-example/primitives.html), every value is _owned_
```

Comment:
I think this is related to the `Copy` trait, not necessarily the primitive types (although they all implement `Copy`, making it a great example).